src/AppInstallerCLICore/Workflows/InstallFlow.cpp

```diff
@@ -408,12 +409,41 @@ namespace AppInstaller::CLI::Workflow
         context <<
             Workflow::SelectInstaller <<
             Workflow::EnsureApplicableInstaller <<
+            Workflow::ManageDependencies <<
             Workflow::InstallPackageInstaller;
     }
 
+    void ManageDependencies(Execution::Context& context)
+    {
+        if (Settings::ExperimentalFeature::IsEnabled(Settings::ExperimentalFeature::Feature::Dependencies))
+        {
+            const auto& installer = context.Get<Execution::Data::Installer>();
+            if (installer && installer->Dependencies.HasAny())

```

Comment:
All dependencies tasks should separate creating a list of dependencies and using that list.  I would not expect to see a task that both adds the list of dependencies and then later consumes them (or calls another task that consumes them).  All of the top level flows should be able to do something like:
```
context << DetermineDependencies << ActOnDependencies;
```
Where the "determine" and "act on" tasks may be different for each flow, but always completely independent from each other.

Reply:
Agree but for import (InstallMultiple) it's hard to divide as the task is not completely divided into workflows.